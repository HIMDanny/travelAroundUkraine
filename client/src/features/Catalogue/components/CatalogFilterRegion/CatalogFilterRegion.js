import React, { useState } from 'react';
import { Box, Checkbox, FormControlLabel } from '@mui/material';
import { FilterAccordion } from '..';

const regions = ['region1', 'region2', 'region3', 'region4'];

const CatalogFilterRegion = () => {
  const [checked, setChecked] = useState(Array(regions.length).fill(false));

  const isAllChecked = checked.every((el) => el === true);

  const handleChangeAll = () => {
    setChecked((current) => {
      const newChecked = [...current];
      if (!isAllChecked) {
        newChecked.fill(true);
      } else {
        newChecked.fill(false);
      }

      return newChecked;
    });
  };

  const handleChange = (event) => {
    const index = regions.indexOf(event.target.id);
    setChecked((current) => {
      const newChecked = [...current];
      newChecked[index] = !newChecked[index];
      return newChecked;
    });
  };

  const children = (
    <Box sx={{ display: 'flex', flexDirection: 'column', ml: 3 }}>
      {regions.map((el) => (
        <FormControlLabel
          key={el}
          label={el}
          control={<Checkbox id={el} checked={checked[regions.indexOf(el)]} onChange={handleChange} />}
        />
      ))}
    </Box>
  );

  return (
    <FilterAccordion title="Region">
      <FormControlLabel label="All regions" control={<Checkbox checked={isAllChecked} onChange={handleChangeAll} />} />
      {children}
    </FilterAccordion>
  );
};

export default CatalogFilterRegion;
