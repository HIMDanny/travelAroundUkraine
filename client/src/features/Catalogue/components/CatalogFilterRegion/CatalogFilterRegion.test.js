import { render } from '@testing-library/react';
import CatalogFilterRegion from './CatalogFilterRegion';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Checkbox', () => ({ children }) => <span>{children}</span>);
jest.mock('@mui/material/FormControlLabel', () => ({ children }) => <label>{children}</label>);

jest.mock('../FilterAccordion/FilterAccordion', () => ({ children }) => <div>{children}</div>);

describe('CatalogFilterRegion snapshot test', () => {
  test('should CatalogFilterRegion match snapshot', () => {
    const { asFragment } = render(<CatalogFilterRegion />);
    expect(asFragment()).toMatchSnapshot();
  });
});
