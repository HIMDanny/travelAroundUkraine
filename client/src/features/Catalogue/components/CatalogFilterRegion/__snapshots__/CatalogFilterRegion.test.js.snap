// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilterRegion snapshot test should CatalogFilterRegion match snapshot 1`] = `
<DocumentFragment>
  <div>
    <label />
    <div>
      <label />
      <label />
      <label />
      <label />
    </div>
  </div>
</DocumentFragment>
`;
